al_aws_collector.js

```diff
@@ -294,13 +344,57 @@ class AlAwsCollector {
             function(asyncCallback) {
                 collector._formatFun(event, context, asyncCallback);
             },
-            function(formatedData, asyncCallback) {
-                collector.send(formatedData, asyncCallback);
+            function(formattedData, compress, asyncCallback) {
+                var args = Array.from(arguments);
+                asyncCallback = args.pop();
+                formattedData = args.shift();
+                if (args.length > 0) {
+                    compress = args.shift();
+                } else {
+                    compress = true;
+                }
```

Comment:
I know this is how we do this sort of thing in other places, but there are nicer ways to handle functions of variable arity in JS.

I mean it would be simpler to just check if the arguments array is length 2 and typecheck the last argument to see if its a function.

This code work in other places so I'm ok to leave it, but I'm mentioning it because there are simpler ways of doing it.

Reply:
I'd really appreciate a code sample here =)

Reply:
Yep. Sorry that would be good.

something like this.
``` javascript
// if there are only two arguments and the second (ie compress) is a function, it must be the callback. Assign it to the asyncCallback variable
// if you wan to be extra safe, we can check if asyncCallback is undefined
if(arguments.length === 2 && typeof compress === "function"){
    asyncCallback = compress
    compress = true
} 
```

here is a test is did in the node console:
```
> function testCall(data, compress, callback){
...     if(arguments.length === 2 && typeof compress === "function"){
.....         callback = compress
.....         compress = true
.....     }
...     callback(compress)
... }
undefined
> testCall("stuff", false, (e) => console.log(e))
false
undefined
> testCall("stuff", (e) => console.log(e))
true
undefined
```

References
console.account.product.dev.alertlogic.com/users/api/index.html#/
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof